### PR TITLE
feat: twitter view for latest tweet 

### PR DESCRIPTION
**Pull request by sracca**
- currently only handles direct tweets from mojotech (i.e. no replies, no retweets) 
- does not support multiple images, only takes first one from tweet for preview.
- If media/content is displayed, previous tweets are not seen.
- IMPORTANT: Please update env.local with Helios Twitter tokens or else it will not work. (E.g. log into Twitter Devlopers under Helios acount, go under 'Apps' to 'Helios' then find under 'Tokens & Permissions')
Link
Screen Shot 2019-06-24 at 3 42 32 PM
Photo
Screen Shot 2019-06-26 at 1 07 56 PM
Text
Screen Shot 2019-06-20 at 3 31 35 PM

### CODE REVIEW

File: app/javascript/components/widgets/twitter/tweet_profile.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const TwitterProfile = ({ dateCreated }) => {
+  return (
+    <ProfileWrapper>
+      <div style={{ marginRight: '24px' }}>
+        <img src={twitterIcon} alt="twitter profile" />
+      </div>
+      <ProfileInfoWrapper>
+        <ProfileHeader>MojoTech</ProfileHeader>
+        <div style={{ display: 'flex', flexDirection: 'row' }}>
+          <ProfileSub style={{ marginRight: '30px' }}>@MojoTech</ProfileSub>
```
Comment (theroyalblah) [inline app/javascript/components/widgets/twitter/tweet_profile.jsx]: Maybe a line like this to get this more like the designs?
`<ProfileSub> • </ProfileSub>`
Screen Shot 2019-06-26 at 1 58 35 PM

File: app/javascript/components/widgets/twitter/panel.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+import { Query } from 'react-apollo';
+import { LoadingMessage, DisconnectedMessage } from '@messages/message';
+import styled from 'styled-components';
+import { colors, fontSizes } from '@lib/theme';
+import { parseMonthDate } from '../../../lib/datetime';
+import TwitterProfile from './tweet_profile';
+import TweetStats from './tweet_stats';
+import TweetBody from './tweet_body';
```
Comment (theroyalblah) [inline app/javascript/components/widgets/twitter/panel.jsx]: This should probably use absolute imports
`import TweetBody from '@twitter/tweet_body';`

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 3]
```diff
@@ -3,40 +3,79 @@ import PropTypes from 'prop-types';
...
-          color: `${colors.white}`,
-        }}
-        searchWords={getMentionsAndTags(displayText)}
-        textToHighlight={displayText}
-      />
-    </TweetText>
+    <React.Fragment>
+      <div>
+        {/* just content, no text */}
+        {typeof displayText === 'undefined' ? null : (
```
Comment (theroyalblah) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: I'm confused about the comment 
> {/* just content, no text */}

Isn't the below code the Twitter text?

File: .gitignore [hunk 4]
```diff
@@ -42,3 +42,5 @@ yarn-debug.log*
 /.yarnrc
 
 .DS_Store
+
+*.rdb
```
Comment (theroyalblah) [inline .gitignore]: No newline at the end of the file here. Also, I'm not sure adding dump.rdb needs to be part of this PR.

File: app/javascript/components/widgets/twitter/panel.jsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+const TweetDivider = styled.div`
+  margin-top: 81px;
+  color: #959292;
+  display: flex;
+  margin-bottom: 16px;
+  flex-wrap: no-wrap;
+  justify-content: space-between;
+  flex-direction: column;
+  width: 45vw;
+  padding-left: 100px;
```
Comment (theroyalblah) [inline app/javascript/components/widgets/twitter/panel.jsx]: Probably should be a margin instead of padding

File: app/javascript/schema.json [hunk 6]
```diff
@@ -116,6 +116,32 @@
               },
               "isDeprecated": false,
               "deprecationReason": null
+            },
+            {
+              "name": "tweets",
```
Comment (cbarber) [inline app/javascript/schema.json]: This schema update happens after the initial schema is introduced in the other PR. I think it makes sense to just create the backend with your final version of the schema since you're introducing a new feature.

File: app/lib/clients/twitter_client.rb [hunk 7]
```diff
@@ -11,11 +11,26 @@ def self.authenticate
     client
   end
 
+  TwitterQuery = Struct.new(:created_at, :text, :favorite_count, :retweet_count, :reply_count)
+
+  def self.reply_count(id, client)
+    replies = client.search("to:mojotech ", since_id: id, include_entities: true)
+    .select { |res| res.in_reply_to_status_id == id }
```
Comment (cbarber) [inline app/lib/clients/twitter_client.rb]: This indentation is off. I'd also expect `.search(....)` to be on a new line as well. Is rubocop not enforcing this?

File: app/lib/clients/twitter_client.rb [hunk 8]
```diff
@@ -11,11 +11,26 @@ def self.authenticate
     client
   end
 
+  TwitterQuery = Struct.new(:created_at, :text, :favorite_count, :retweet_count, :reply_count)
+
+  def self.reply_count(id, client)
+    replies = client.search("to:mojotech ", since_id: id, include_entities: true)
```
Comment (cbarber) [inline app/lib/clients/twitter_client.rb]: Is the space after `mojotech` necessary? Some API quirk?

File: app/lib/clients/twitter_client.rb [hunk 9]
```diff
@@ -11,11 +11,26 @@ def self.authenticate
...
+  
   def self.latest_tweet
     client = authenticate
     # API gets count amt, then filters out rts and replies
     # This guarantees there will always be a tweet
-    tweet = client.user_timeline('mojotech', count: 5, include_rts: false, exclude_replies: true).first
-    tweet
+    latest = client.user_timeline('mojotech', count: 5, include_rts: false, exclude_replies: true, tweet_mode: 'extended').first
+
+    tweet_text = latest.attrs[:full_text]
```
Comment (cbarber) [inline app/lib/clients/twitter_client.rb]: Oh derp. I didn't realize they exposed `attrs` publicly. Good call!

File: app/lib/clients/twitter_client.rb [hunk 10]
```diff
@@ -11,11 +11,26 @@ def self.authenticate
...
+    replies.count
+  end
+  
   def self.latest_tweet
     client = authenticate
     # API gets count amt, then filters out rts and replies
     # This guarantees there will always be a tweet
-    tweet = client.user_timeline('mojotech', count: 5, include_rts: false, exclude_replies: true).first
-    tweet
+    latest = client.user_timeline('mojotech', count: 5, include_rts: false, exclude_replies: true, tweet_mode: 'extended').first
```
Comment (cbarber) [inline app/lib/clients/twitter_client.rb]: Could we potentially simplify our API calls if we set `exclude_replies: false` here and aggregating the reply count for this?

File: db/schema.rb [hunk 11]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_06_18_200946) do
+ActiveRecord::Schema.define(version: 2019_06_04_172236) do
```
Comment (cbarber) [inline db/schema.rb]: Shouldn't see this change here without a schema change

File: spec/graphql/tweet_spec.rb [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+describe Helios2Schema do
+  let(:context) { {} }
+  let(:variables) { {} }
+
+  before {
+    allow(ENV).to receive(:[]).with('TWITTER_CONSUMER_KEY') { 'uPMWuw6MsuNb2TXjPcYKZdttQ' }
+    allow(ENV).to receive(:[]).with('TWITTER_CONSUMER_SECRET') { '6wRZEjAJQOOVhtOvKKVJx36VRwUYCurkYCWxu6h0N65vsZLw1w' }
+    allow(ENV).to receive(:[]).with('TWITTER_ACCESS_TOKEN') { '1131245136149721088-l41M9JVBW58HdAhZ9SDILDEpmsQ1SU' }
+    allow(ENV).to receive(:[]).with('TWITTER_ACCESS_SECRET') { 'Hf0ar5EsBYrFV5oA9OhQAH9IbrbkwBQZxhTvEM1dWDhii' }
```
Comment (cbarber) [inline spec/graphql/tweet_spec.rb]: Are these real credentials? Should be removed right away.

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 13]
```diff
@@ -13,9 +14,25 @@ const getStatusLink = text => {
   return allWords.filter(word => word.includes('https://t.co/'))[0];
 };
 
+const getMentionsAndTags = text => {
+  const allWords = text.split(/[\s,\n]+/);
+  return allWords.filter(word => word.includes('@') || word.includes('#'));
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: We should have some test coverage on this. Could potentially consolidate the includes into the regex. Is this a valid mention `foor@bar`? Is this a valid tag `foo#bar`?

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 14]
```diff
@@ -13,9 +14,25 @@ const getStatusLink = text => {
...
+
 const TweetBody = ({ text }) => {
   const displayText = text.replace(getStatusLink(text), ' ');
-  return <TweetText>{displayText}</TweetText>;
+  return (
+    <TweetText>
+      <Highlighter
+        highlightStyle={{
+          backgroundColor: 'transparent',
+          color: `${colors.white}`,
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: Could this use styled-components instead of the inline style?

File: package.json [hunk 15]
```diff
@@ -4,6 +4,7 @@
   "dependencies": {
     "@babel/preset-env": "^7.4.5",
     "@babel/preset-react": "^7.0.0",
+    "@microlink/react": "^4.0.1",
```
Comment (cbarber) [inline package.json]: I feel like this commit is sub-atomic.

File: app/javascript/components/widgets/twitter/panel.jsx [hunk 16]
```diff
@@ -57,24 +60,36 @@ const SubscribedTwitter = ({ tweets }) => {
...
         <TweetStats
           favorites={favoriteCount}
           retweets={retweetCount}
           replies={replyCount}
         />
       </TweetWrapper>
-      <div>
+      <div
+        style={{
+          visibility: noContent ? 'visible' : 'hidden',
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/panel.jsx]: Prefer styled-components to inline styles.

File: app/javascript/components/widgets/twitter/tweet_content.jsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+        url={linkUrl}
+        style={{
+          fontFamily: `${fonts.regular}`,
+          height: '487px',
+          maxWidth: '100%',
+          color: `${colors.white}`,
+          backgroundColor: `${colors.black}`,
+          borderRadius: '6px',
+          borderCollapse: 'separate',
+          border: 'solid 1px #333333',
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_content.jsx]: Same here. Could this be resolved with a containing styled component? Or by children / a component prop?

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 3 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: Sorry I think I didn't write it clearer: if there's no text, render null.

Comment (theroyalblah) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: ahhh that makes sense, thanks!

File: app/lib/clients/twitter_client.rb (hunk 9 above)
Comment (sracca) [reply app/lib/clients/twitter_client.rb]: Hahah yeah I stumbled upon it trying to fix some stuff.

File: app/lib/clients/twitter_client.rb (hunk 10 above)
Comment (sracca) [reply app/lib/clients/twitter_client.rb]: No, so when they are getting a timeline, they are getting replies Mojotech itself has sent (i.e. what's on their timeline). I unfortunately have to count up the amount of replies *specifically* to the latest tweet, as it's not a data piece they give themselves.

File: app/javascript/components/widgets/twitter/tweet_content.jsx (hunk 17 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_content.jsx]: I tried using styled-components on the Microlink and for some reason the API wouldn't accept the styling (it was in the docS). I'm not super familiar on how to style API components with styled so I'm not sure if it was me or whatever

**Comment by sracca** [comment]
@cbarber I fixed the spec tests, as well as refactored a few minor bits of code

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+import { fontSizes } from '../../../lib/theme';
+
+const TweetText = styled.div`
+  color: #959292;
+  font-size: ${fontSizes.large};
+  margin-bottom: 29px;
+`;
+const getStatusLink = text => {
+  const allWords = text.split(/[\s,\n]+/);
+  return allWords.filter(word => word.includes('https://t.co/'))[0];
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: I think this could be simplified and should have a unit test. Specifically the regex could capture the first match using `https://t.co/` and your separator constraints.

File: app/javascript/components/widgets/twitter/tweet_profile.jsx [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
+const TwitterProfile = ({ dateCreated }) => {
+  return (
+    <ProfileWrapper>
+      <ProfileIcon src={twitterIcon} alt="twitter profile" />
+      <ProfileInfoWrapper>
+        <ProfileHeader>MojoTech</ProfileHeader>
+        <div style={{ display: 'flex', flexDirection: 'row' }}>
+          <ProfileSub>@MojoTech</ProfileSub>
+          <ProfileSub style={{ marginLeft: '15px', marginRight: '15px' }}>
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_profile.jsx]: Avoid inline styles here and above with styled component. Could the spaces around `•` be replaced with left/right padding?

File: app/javascript/components/widgets/twitter/tweet_stats.jsx [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+
+const IconLabel = styled.div`
+  color: ${colors.white};
+  font-size: ${fontSizes.small};
+  opacity: 0.5;
+  margin-left: 8px;
+`;
+
+const interactionVis = count => {
+  if (count === 0) return 'none';
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_stats.jsx]: Nit: this could be a ternary.

File: app/javascript/components/widgets/twitter/tweet_stats.jsx [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+  return '';
+};
+
+const StatsIcon = ({ icon, count }) => (
+  <IconWrapper>
+    <img src={icon} alt="stat_icon" />
+    <IconLabel
+      style={{
+        display: interactionVis(count),
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_stats.jsx]: You can handle this as a prop in the styled component.

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 22]
```diff
@@ -1,21 +1,35 @@
...
+
+const highlighterStyle = {
+  backgroundColor: 'transparent',
+  color: `${colors.white}`,
+};
+
+const getMentionsAndTags = text => {
   const allWords = text.split(/[\s,\n]+/);
-  return allWords.filter(word => word.includes('https://t.co/'))[0];
+  return allWords.filter(word => word.includes('@') || word.includes('#'));
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: I'd like to see a unit test for what is/isn't a mention or tag.

File: app/javascript/components/widgets/twitter/tweet_content.jsx [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+
+const TweetText = styled.div`
+  color: #959292;
+  font-size: ${fontSizes.large};
+  margin-bottom: 29px;
+`;
+
+const getMentionsAndTags = text => {
+  const allWords = text.split(/[\s,\n]+/);
+  return allWords.filter(word => word.startsWith('@') || word.startsWith('#'));
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_content.jsx]: Some repetition with this function. Better to extract it into a utils.js in the folder and share.

File: app/javascript/components/widgets/twitter/tweet_content.jsx [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+      >
+        {linkUrl}
+      </TweetText>
+    </div>
+    <TweetMedia mediaUrl={mediaUrl} linkUrl={undefined} />
+  </React.Fragment>
+);
+
+export const OneMediaType = (displayText, mediaUrl, linkUrl) => (
+  <React.Fragment>
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_content.jsx]: Prefer `<>` over `React.Fragment`

File: app/javascript/components/widgets/twitter/tweet_content.jsx [hunk 25]
```diff
@@ -0,0 +1,124 @@
...
+            backgroundColor: 'transparent',
+            color: `${colors.white}`,
+          }}
+          searchWords={getMentionsAndTags(displayText)}
+          textToHighlight={displayText}
+        />
+      </TweetText>
+    </div>
+    <TweetMedia
+      mediaUrl={mediaUrl === null ? undefined : mediaUrl}
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_content.jsx]: Hrm. Why convert `null` to `undefined` here?

File: app/javascript/components/widgets/twitter/tweet_content.jsx [hunk 26]
```diff
@@ -0,0 +1,124 @@
...
+      linkUrl={linkUrl === null ? undefined : linkUrl}
+    />
+  </React.Fragment>
+);
+
+const TweetMedia = ({ mediaUrl, linkUrl }) => (
+  <ContentWrapper>
+    {mediaUrl !== null ? <ContentImage src={mediaUrl} alt="twitter" /> : null}
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_content.jsx]: Prefer `condition && <Component />` to `condition ? <Component /> : null`. Was this a react native idiom?

File: spec/graphql/tweet_spec.rb [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+
+      it 'will return a query with valid data' do
+        latest = result["data"]["tweets"].first
+        expect(latest["createdAt"]).to eq("2019-06-28 18:01:35 UTC")
+        expect(latest["text"]).to eq("Spending the afternoon with our friends at @HumaneBoulder! ")
+        expect(latest["favoriteCount"]).to eq(0)
+        expect(latest["retweetCount"]).to eq(0)
+        expect(latest["replyCount"]).to eq(0)
+        expect(latest["media"]).to eq("http://pbs.twimg.com/media/D-KsunCW4AA7uz8.jpg")
+        expect(latest["link"]).to eq(nil)
```
Comment (cbarber) [inline spec/graphql/tweet_spec.rb]: Nice! :100:

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 28]
```diff
@@ -1,39 +1,53 @@
...
+      linkUrl={linkUrl === null ? undefined : linkUrl}
+    />
+  </React.Fragment>
+);
 
-const getMentionsAndTags = text => {
-  const allWords = text.split(/[\s,\n]+/);
-  return allWords.filter(word => word.includes('@') || word.includes('#'));
+const TweetBody = ({ text, mediaUrl, linkUrl }) => {
+  if (mediaUrl === null && linkUrl === null) return <TweetText text={text} />;
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: Prefer using brackets for these blocks.

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 18 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: I'm not really sure, but I removed that from the history a while ago I think?

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 28 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: Do you mean the conditionals? Because if so, they're against the current eslint rules

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 18 above)
Comment (cbarber) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: The unit test or the regex?

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 28 above)
Comment (cbarber) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: No errors on my end. What does your editor show you?
condition-single-lines

Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: I'm not getting errors but whenever I hit save, prettier removes them. I'll check my rules

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 18 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: Oh this whole function. I now have it so it already does it on the back-end when getting it from the client.

**Comment by sracca** [comment]
@cbarber I added unit tests to getTagsAndMentions, as well as removed replies and fixed the other issues.

File: app/javascript/components/widgets/twitter/panel.jsx [hunk 29]
```diff
@@ -1,3 +1,95 @@
...
+const SubscribedTwitter = ({ tweets }) => {
+  const latestTweet = tweets[0];
+  const { createdAt, text, favoriteCount, retweetCount } = latestTweet;
+
+  return (
+    <Wrapper>
+      <TweetWrapper>
+        <TwitterProfile dateCreated={parseMonthDate(createdAt)} />
+        <TweetBody text={text} />
+        <TweetStats favorites={favoriteCount} retweets={retweetCount} />
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/panel.jsx]: Nice job breaking these up into small components :100:

File: app/javascript/components/widgets/twitter/panel.jsx [hunk 30]
```diff
@@ -1,3 +1,95 @@
...
+    {({ loading, error, data }) => {
+      if (loading) {
+        return <LoadingMessage />;
+      }
+
+      if (error) {
+        return <DisconnectedMessage />;
+      }
+
+      return <SubscribedTwitter tweets={data.tweets} />;
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/panel.jsx]: Technically not `Subscribed` since there is no graphql subscription.

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+  font-size: ${fontSizes.large};
+  margin-bottom: 29px;
+`;
+const getStatusLink = text => {
+  const allWords = text.split(/[\s,\n]+/);
+  return allWords.filter(word => word.includes('https://t.co/'))[0];
+};
+
+const TweetBody = ({ text }) => {
+  const displayText = text.replace(getStatusLink(text), ' ');
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: You mentioned [1] that getStatusLink replacement is happening in the backend now but looks like its still used:
https://github.com/mojotech/helios2/pull/130/commits/e95b3bedaddd1d5059074dc55a3251f0084f5edc#r302201079

File: app/javascript/components/widgets/twitter/tweet_stats.jsx [hunk 32]
```diff
@@ -0,0 +1,57 @@
...
+  </IconWrapper>
+);
+
+StatsIcon.propTypes = {
+  icon: PropTypes.string.isRequired,
+  count: PropTypes.number.isRequired,
+};
+
+const TweetStats = ({ favorites, retweets }) => {
+  return (
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_stats.jsx]: Nit: prefer `=> ( ..... )` shorthand for only return functions

File: app/javascript/lib/util.test.js [hunk 33]
```diff
@@ -32,3 +32,28 @@ describe('#isPresent', () => {
...
+    expect(getMentionsAndTags('The quick brown fox@')).toEqual([]);
+    expect(getMentionsAndTags('The quick brown fo@x')).toEqual([]);
+  });
+  it('handles a combination', () => {
+    expect(getMentionsAndTags('The quick #brown @fox')).toEqual([
+      '#brown',
+      '@fox',
+    ]);
+    expect(getMentionsAndTags('The quick br#own fox@')).toEqual([]);
+    expect(getMentionsAndTags('The quick #brown fo@x')).toEqual(['#brown']);
```
Comment (cbarber) [inline app/javascript/lib/util.test.js]: :100: nice coverage of edge cases

File: app/javascript/components/widgets/twitter/tweet_body.jsx [hunk 34]
```diff
@@ -1,35 +1,53 @@
...
+  <>
+    <div>
+      <TweetText text={displayText} />
+      <LinkText>{linkUrl}</LinkText>
+    </div>
+    <TweetMedia mediaUrl={mediaUrl} linkUrl={undefined} />
+  </>
+);
+
+export const OneMediaType = (displayText, mediaUrl, linkUrl) => (
```
Comment (cbarber) [inline app/javascript/components/widgets/twitter/tweet_body.jsx]: Nit: these could just be components. Doesn't look like the export is ever used.

File: app/javascript/components/widgets/twitter/panel.jsx (hunk 30 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/panel.jsx]: Do you think I should rename?

File: app/javascript/components/widgets/twitter/tweet_body.jsx (hunk 31 above)
Comment (sracca) [reply app/javascript/components/widgets/twitter/tweet_body.jsx]: Ah okay, I think it's snuck it's way into an older commit so I didn't see it. I'll remove it.

**Comment by sracca** [comment]
@craigb all fixed!
